src/envoy/http/authn/http_filter.cc

```diff
@@ -14,23 +14,30 @@
  */
 
 #include "src/envoy/http/authn/http_filter.h"
+#include <memory>
+#include "common/http/utility.h"
 #include "src/envoy/http/authn/mtls_authentication.h"
-#include "src/envoy/utils/utils.h"
 
 namespace Envoy {
 namespace Http {
 
 AuthenticationFilter::AuthenticationFilter(
-    const istio::authentication::v1alpha1::Policy& config)
-    : config_(config) {}
+    const istio::authentication::v1alpha1::Policy& config,
+    Upstream::ClusterManager& cm,
+    std::shared_ptr<Http::JwtAuth::JwtAuthStoreFactory> jwt_store_factory)
+    : config_(config), cm_(cm), jwt_store_factory_(jwt_store_factory) {}
```

Comment:
I suggest you pass in  

JwtAuthStore*

I don't like to pass in factory into the filter code.

Reply:
Done.